### DISCUSSION TITLE
Weekly timeseries

### DISCUSSION THREAD

**Discussion by ghost** [autogluon/autogluon#3464] [General]
Hi, 
can i use weekly frequency with autogluon in order to predict next value for such timeseries :
02/01/2023 -> 09/01/2023 -> 16/01/2023 -> ...
do i have to fill intermediate values with fill_missing_values() method ?
Best regards.

**Comment by shchur** [comment]
Hi,

`TimeSeriesPredictor` can directly handle such data. You can verify that you have regular timestamps with weekly frequency with 
```python
ts_df = TimeSeriesDataFrame(...)
print(ts_df.freq)
```
If you see output like "W-SUN", "W-MON", "W-TUE", ... here, then you can just pass the data to the predictor.

If you see `None`, it means that the timestamps are irregular. You can convert the data to a regular weekly frequency with
```python
ts_df = ts_df.to_regular_index(freq="W")
ts_df = ts_df.fill_missing_values()
```

**Reply by ghost** [reply]
Thanks.
